platform/reference/migrations/4-0-migration.mdx

```diff
@@ -0,0 +1,129 @@
+part of the upgrade command. Since future upgrades reuse the values of the configuration, you only need to apply it once.
+
+### Appending the project prefix to upgrade commands
+
+<Tabs
+  defaultValue="cli"
+  values={[
+    { label: 'CLI', value: 'cli', },
+    { label: 'helm', value: 'helm', },
+  ]
+}>
+<TabItem value="cli">
+
+To upgrade the Platform from v3 to v4 via vCluster CLI, run. Update `$PLATFORM_VERSION` with a valid vCluster Platform version.
+
+:::info Plaftorm Namsespace
+The following steps assume the existing installation is in the default namespace `loft`. The default namespace for v4 is vcluster-platform, so we specify the namespace in the steps.
+:::
+
+```bash
+PLATFORM_NAMESPACE="loft" # update if the platform is installed to a different namespace
+PLATFORM_VERSION="" # specify a version or it will upgrade to latest stable
+
+vcluster platform start --upgrade --namespace $PLATFORM_NAMESPACE --version=$PLATFORM_VERSION --values=<(cat <<EOF
+config:
+  projectNamespacePrefix: loft-p-
+EOF
+)
+```
+
+</TabItem>
+<TabItem value="helm">
+
+To upgrade vCluster Platform from v3 to v4 via `helm`, run. Update `$PLATFORM_VERSION` with a valid vCluster Platform version.
+
+```bash
+PLATFORM_RELEASE_NAME="loft" # update if the Helm release has a different name
+PLATFORM_NAMESPACE="loft" # update if the platform is installed to a different namespace
+PLATFORM_VERSION="" # specify a version or it will upgrade to latest stable
+
+helm upgrade $PLATFORM_RELEASE_NAME vcluster-platform -n $PLATFORM_NAMESPACE --repository-config '' --repo https://charts.loft.sh \
+  --version $PLATFORM_VERSION \
+  --reuse-values \
+  -f <(cat <<EOF
+config:
+  projectNamespacePrefix: loft-p-
+EOF
+)
+```
+
+</TabItem>
+</Tabs>
+
+### Adding the project prefix to your values file
+
+If you upgrade the Platform with another method, an alternative method is to extend your config file and include the project prefix requirement for this upgrade.
+
+```
+config:
+  projectNamespacePrefix: loft-p-
+```
+
+## Troubleshooting Tips 
+
+### Forgot the Project Prefix
+
+If you accidentially performed the upgrade without setting the `projectNamespacePrefix`, then the pod of the platform will be in a `CrashLoopBackoff` with a log similiar to:
+
+```bash
+cmd/main.go:107 error executing root command    {"component": "loft", "error": "init (4): set default project namespace prefix: seems like you have upgraded the platform from an earlier version that uses 'loft-p-' as a project namespace prefix. This has been changed to be 'p-' in the current version. Please set 'projectNamespacePrefix: loft-p-' in the platform config to get rid of this error"}
+```
+
+In that case wait until the Pod starts up again and retry the above upgrade command by passing along a vcluster-platform.yaml file that contains the config change.
```

Comment:
May say this instead?

`retry the above upgrade command by passing along a vcluster-platform.yaml file that contains the config change for projectNamespacePrefix`

Reply:
I was waiting for some confirmation on what to do, so I've completely changed the sentence.